centreon/www/include/configuration/configObject/contact/formContact.php

```diff
@@ -479,6 +479,147 @@
+                    $parentsLvl[$parentLvl1][$parentLvl2][] = $page;
+                }
+            }
+        }
+
+        return $parentsLvl;
+    };
+
+    /**
+     * Check if at least one child can be shown
+     */
+    $oneChildCanBeShown = function () use (&$childrenLvl3, &$translatedPages): bool {
+        $isCanBeShow = false;
+        foreach ($childrenLvl3 as $topologyPage) {
+            if ($translatedPages[$topologyPage]['show']) {
+                $isCanBeShow = true;
+                break;
+            }
+        }
+        return $isCanBeShow;
+    };
+
+    $topologies = $createTopologyTree($acls);
+
+    /**
+     * Retrieve the name of all topologies available for this user
+     */
+    $aclTopologies = $pearDB->query(
+        "SELECT topology_page, topology_name, topology_show "
+        . "FROM topology "
+        . "WHERE topology_page IN ($aclUser)"
+    );
+
+    $translatedPages = [];
+
+    while ($topology = $aclTopologies->fetch(\PDO::FETCH_ASSOC)) {
+        $translatedPages[$topology['topology_page']] = [
+            'i18n' => _($topology['topology_name']),
+            'show' => ((int)$topology['topology_show'] === 1)
+        ];
+    }
+
+    /**
+     * Create flat tree for menu with the topologies names
+     * [item1Id] = menu1 > submenu1 > item1
+     * [item2Id] = menu2 > submenu2 > item2
+     */
+    foreach ($topologies as $parentLvl1 => $childrenLvl2) {
+        $parentNameLvl1 = $translatedPages[$parentLvl1]['i18n'];
+        foreach ($childrenLvl2 as $parentLvl2 => $childrenLvl3) {
+            $parentNameLvl2 = $translatedPages[$parentLvl2]['i18n'];
+            $isThirdLevelMenu = false;
+            $parentLvl3 = null;
+
+            if ($oneChildCanBeShown()) {
+                /**
+                 * There is at least one child that can be shown then we can
+                 * process the third level
+                 */
+                foreach ($childrenLvl3 as $parentLvl3) {
+                    if ($translatedPages[$parentLvl3]['show']) {
+                        $parentNameLvl3 = $translatedPages[$parentLvl3]['i18n'];
+
+                        if ($parentNameLvl2 === $parentNameLvl3) {
+                            /**
+                             * The name between lvl2 and lvl3 are equals.
+                             * We keep only lvl1 and lvl3
+                             */
+                            $pages[$parentLvl3] = $parentNameLvl1 . ' > '
+                                . $parentNameLvl3;
+                        } else {
+                            $pages[$parentLvl3] = $parentNameLvl1 . ' > '
+                                . $parentNameLvl2 . ' > '
+                                . $parentNameLvl3;
+                        }
+                    }
+                }
+
+                $isThirdLevelMenu = true;
+            }
+
+            // select parent from level 2 if level 3 is missing
+            $pageId = $parentLvl3 ?: $parentLvl2;
```

Comment:
Maybe prefer coalesce over elvis operator to be more precise ?
Elvis will avoid the integer `'0'`, but not the coalesce.
```suggestion
            $pageId = $parentLvl3 ?? $parentLvl2;
```